ignite/metrics/accuracy.py

```diff
@@ -6,47 +6,98 @@
+        else:
+            self._threshold = threshold_function
+        self._updated = False
+        super(_BaseClassification, self).__init__(output_transform=output_transform)
+
+    def _check_shape(self, output):
         y_pred, y = output
 
-        if not (y.ndimension() == y_pred.ndimension() or y.ndimension() + 1 == y_pred.ndimension()):
-            raise ValueError("y must have shape of (batch_size, ...) and y_pred "
-                             "must have shape of (batch_size, num_classes, ...) or (batch_size, ...).")
-
         if y.ndimension() > 1 and y.shape[1] == 1:
             y = y.squeeze(dim=1)
 
         if y_pred.ndimension() > 1 and y_pred.shape[1] == 1:
             y_pred = y_pred.squeeze(dim=1)
 
+        if not (y.ndimension() == y_pred.ndimension() or y.ndimension() + 1 == y_pred.ndimension()):
+            raise ValueError("y must have shape of (batch_size, ...) and y_pred must have "
+                             "shape of (batch_size, num_categories, ...) or (batch_size, ...).")
+
         y_shape = y.shape
         y_pred_shape = y_pred.shape
 
         if y.ndimension() + 1 == y_pred.ndimension():
-            y_pred_shape = (y_pred_shape[0], ) + y_pred_shape[2:]
+            y_pred_shape = (y_pred_shape[0],) + y_pred_shape[2:]
 
         if not (y_shape == y_pred_shape):
             raise ValueError("y and y_pred must have compatible shapes.")
 
-        if y_pred.ndimension() == y.ndimension():
-            # Maps Binary Case to Categorical Case with 2 classes
-            y_pred = y_pred.unsqueeze(dim=1)
-            y_pred = torch.cat([1.0 - y_pred, y_pred], dim=1)
+        return y_pred, y
+
+    def _check_type(self, output):
+        y_pred, y = output
+
+        if y.ndimension() + 1 == y_pred.ndimension():
+            update_type = 'multiclass'
+        elif (y_pred.shape == y.shape) or y_pred.shape[1] == 1:
+            update_type = 'binary'
+            if not self._updated and self._threshold is None:
+                self._threshold = torch.round
+        else:
+            raise TypeError('Invalid shapes of y (shape={}) and y_pred (shape={}), check documentation'
+                            ' for expected shapes of y and y_pred.'.format(y.shape, y_pred.shape))
+        if not self._updated:
+            self._type = update_type
+            self._updated = True
+        else:
+            if self._type != update_type:
+                raise TypeError('update_type has changed from {} to {}.'.format(self._type, update_type))
+
+
+class Accuracy(_BaseClassification):
+    """
+    Calculates the accuracy.
+
+    - `update` must receive output of the form `(y_pred, y)`.
+    - `y_pred` must be in the following shape (batch_size, num_categories, ...) or (batch_size, ...)
+    - `y` must be in the following shape (batch_size, ...)
+    """
+
+    def __init__(self, output_transform=lambda x: x, threshold_function=None):
```

Comment:
Probably, no need of constructor ?

Reply:
But I'm OK with that if this help user to understand better how to construct the object

Reply:
Same as above, removed threshold. Agree about removing constructor. It doesn't serve a purpose.